presto-main/src/main/java/com/facebook/presto/operator/JoinOperatorFactory.java

```diff
@@ -15,8 +15,32 @@
 
 import java.util.Optional;
 
+import static java.util.Objects.requireNonNull;
+
 public interface JoinOperatorFactory
         extends OperatorFactory
 {
-    Optional<OperatorFactory> createOuterOperatorFactory();
+    Optional<OuterOperatorFactoryResult> createOuterOperatorFactory();
+
+    class OuterOperatorFactoryResult
```

Comment:
`OuterOperatorFactoryAndExeuctionStrategy`? ... I am not sure.....

Reply:
Keeping this as is right now. I prefer the name `OuterOperatorFactoryResult` to `OuterOperatorFactoryAndExeuctionStrategy` because it just indicates it's the return type of `createOuterOperatorFactory` and doesn't get into intimate details of exactly what's included, which could potentially change in the future.